### PR TITLE
ENH: add plot_risk_exposures

### PR DESCRIPTION
**Pull request by vikram-narayan**
1. plot risk factor exposures:
  risk factor exposures

2. add flag `pos_in_dollars` to `perf_attrib` indicating whether positions are in dollars or not (for #439)

### CODE REVIEW

File: pyfolio/perf_attrib.py [hunk 1]
```diff
@@ -225,7 +237,8 @@ def plot_factor_contribution_to_perf(exposures, perf_attrib_data, ax=None):
         perf_attrib_data.index,
         [perf_attrib_data[s] for s in chain(perf_attrib_data.iloc[:, :-3],
                                             ['specific_returns'])],
-        labels=list(perf_attrib_data.iloc[:, :-3]) + ['specific returns']
+        labels=list(perf_attrib_data.iloc[:, :-3]) + ['specific returns'],
```
Comment (twiecki) [inline pyfolio/perf_attrib.py]: why are the last three dropped?

File: pyfolio/perf_attrib.py [hunk 2]
```diff
@@ -235,3 +248,26 @@ def plot_factor_contribution_to_perf(exposures, perf_attrib_data, ax=None):
...
+    """
+    Parameters
+    ----------
+    exposures : pd.DataFrame
+        df indexed by datetime, with factors as columns
+        - Example:
+                        momentum  reversal
+            dt
+            2017-01-01 -0.238655  0.077123
+            2017-01-02  0.821872  1.520515
```
Comment (twiecki) [inline pyfolio/perf_attrib.py]: document `ax`

File: pyfolio/perf_attrib.py (hunk 1 above)
Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: the last three columns are `total_returns`, `common_returns`, and `specific_returns`, and common returns are accounted for by each of the individual factor returns

Comment (twiecki) [reply pyfolio/perf_attrib.py]: maybe `perf_attrib_data.drop(['total_returns', ...], axis='columns')` then to make the intent more clear?

File: pyfolio/utils.py [hunk 3]
```diff
@@ -598,3 +598,15 @@ def get_symbol_rets(symbol, start=None, end=None):
     return SETTINGS['returns_func'](symbol,
                                     start=start,
                                     end=end)
+
+
+def set_legend_location(ax):
+    """
+    Put legend in right of plot instead of overlapping with the it.
```
Comment (eigenfoo) [inline pyfolio/utils.py]: Typo `instead of overlapping with the it.`

File: pyfolio/perf_attrib.py [hunk 4]
```diff
@@ -161,6 +180,15 @@ def show_perf_attrib_stats(returns, positions, factor_returns,
 def plot_returns(returns, specific_returns, common_returns, ax=None):
     """
     Plot total, specific, and common returns.
+
+    Parameters
```
Comment (eigenfoo) [inline pyfolio/perf_attrib.py]: Also document `returns`, `specific_returns` and `common_returns`

**Comment by twiecki** [comment]
See feedback from @eigenfoo but looks good otherwise.

File: pyfolio/perf_attrib.py [hunk 5]
```diff
@@ -92,10 +108,13 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings):
...
             2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
     """
-    # convert holdings to percentages, and convert positions to long format
-    positions = get_percent_alloc(positions)
+    if pos_in_dollars:
+        # convert holdings to percentages, and convert positions to long format
+        positions = get_percent_alloc(positions)
+
     # remove cash after normalizing positions
     del positions['cash']
```
[the remark this answers is not in the document]
Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: fixed

File: pyfolio/perf_attrib.py (hunk 4 above)
Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: fixed

File: pyfolio/utils.py (hunk 3 above)
Comment (vikram-narayan) [reply pyfolio/utils.py]: fixed

**Comment by vikram-narayan** [comment]
@yankees714 mind taking a look when you can?

File: pyfolio/perf_attrib.py [hunk 6]
```diff
@@ -17,13 +17,25 @@
...
+# http://phrogz.net/css/distinct-colors.html
+COLORS = [
+    '#f23d3d', '#828c23', '#698c83', '#594080', '#994d4d',
+    '#206380', '#dd39e6', '#cc9999', '#7c8060', '#66adcc',
+    '#6c7dd9', '#8a698c', '#7f6340', '#66cc7a', '#a3abd9',
+    '#d9c0a3', '#bfffcc', '#542699', '#b35986', '#d4e639',
+    '#b380ff', '#e0e6ac', '#a253a6', '#418020', '#ff409f',
+    '#ffa940', '#83ff40', '#3d58f2', '#e3ace6', '#d9a86c',
+    '#2db391'
+]
```
Comment (yankees714) [inline pyfolio/perf_attrib.py]: Would it make sense to put this in `pyfolio.utils`?

File: pyfolio/perf_attrib.py [hunk 7]
```diff
@@ -92,10 +108,13 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings):
             2017-01-01  0.249087  0.935925        1.185012          1.185012
             2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
     """
-    # convert holdings to percentages, and convert positions to long format
-    positions = get_percent_alloc(positions)
+    if pos_in_dollars:
+        # convert holdings to percentages, and convert positions to long format
```
Comment (yankees714) [inline pyfolio/perf_attrib.py]: What does 'convert positions to long format' mean here?

Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: that part of the comment applies to the line `positions = positions.stack()`, it means change it from wide format:
```
                AAPL       TLT       XOM
2017-01-01  0.333333  0.568627  0.098039
2017-01-02  0.188034  0.658120  0.153846
2017-01-03 -0.263158  0.473684  0.526316
```

to long format (everything in one column with a multi index)
```
2017-01-01  AAPL    0.333333
            TLT     0.568627
            XOM     0.098039
2017-01-02  AAPL    0.188034
            TLT     0.658120
            XOM     0.153846
2017-01-03  AAPL   -0.263158
            TLT     0.473684
            XOM     0.526316
```

File: pyfolio/perf_attrib.py (hunk 6 above)
Comment (vikram-narayan) [reply pyfolio/perf_attrib.py]: yup, I'll move it there

**Comment by vikram-narayan** [comment]
I'll squash commits, and then merge

**Comment by vikram-narayan** [comment]
once tests pass, I'll merge

**Pull request merged**